src/layer_shims.cpp

```diff
@@ -583,13 +668,40 @@ XRAPI_ATTR XrResult XRAPI_CALL thisLayer_xrGetActionStateBoolean(XrSession sessi
 	return res;
 }
 
+void recordApplyHapticFeedback(XrHapticActionInfo* hapticActionInfo, XrHapticBaseHeader* hapticBaseHeader)
+{
+       // look up the paths mapped to this action
+       if (auto search = actionBindingMap.find(hapticActionInfo->action); search != actionBindingMap.end())
+       {
+               // iterate over all paths for this action
+               auto paths = actionBindingMap[hapticActionInfo->action];
+               for (auto p : paths)
+               {
+                       // check which path got activated
+                       auto ps = pathToString[p];
+                       auto pss = pathToString[hapticActionInfo->subactionPath];
+                       if (ps.rfind(pss, 0) == 0)
+                       {
+                               // log the path and the data for the action
+                               tracer::traceEntry e = {frameTime, 'h', ps};
+                               tracer::traceApplyHaptic taf = {true};
+                               e.body = taf;
+                               tracer::writeApplyHaptic(e);
+                       }
+               }
+       }
+}
+
 XRAPI_ATTR XrResult XRAPI_CALL thisLayer_xrApplyHapticFeedback(XrSession session, XrHapticActionInfo* hapticActionInfo, XrHapticBaseHeader* hapticBaseHeader) {
     static PFN_xrApplyHapticFeedback nextLayer_xrApplyHapticFeedback = GetNextLayerFunction(xrApplyHapticFeedback);
     // Disable haptic feedback when in replay mode
     if (mode == tracer::REPLAY) {
-        return XR_SUCCESS;
-    }
-    auto res = nextLayer_xrApplyHapticFeedback(session, hapticActionInfo, hapticBaseHeader);
+    	return XR_SUCCESS;
```

Comment:
Strange indent?

Reply:
I think this is github being weird with tabs. As you can see in the rest of the code it apprears too large everywhere. Do you use spaces?

Reply:
To prevent this from happening in the future I've added an `.editorconfig` file on the main branch. If you pull and reformat your code this should all be taken care of.